sphinx/util/inspect.py

```diff
@@ -824,6 +824,24 @@ def signature_from_ast(node: ast.FunctionDef, code: str = '') -> inspect.Signatu
     return inspect.Signature(params, return_annotation=return_annotation)
 
 
+def getclassdoc(obj: Type, allow_inherited: bool = False) -> Optional[str]:
+    """Get the docstring for the class."""
+    from sphinx.ext.autodoc.mock import ismock
+
+    if not allow_inherited:
+        return safe_getattr(obj, '__doc__', None)
+    else:
+        for basecls in getmro(obj)[:-1]:
```

Comment:
This excepts the `object` class. But other built-in classes are not excepted yet. So the docstring of list, dict, str, int and so on are fetched. I think it's an unexpected broken change.

Reply:
Oh...
```
>>> inspect.isbuiltin(list)
False
>>> list.__module__
'builtins'
```

Reply:
Oh...
```
>>> inspect.isbuiltin(list)
False
>>> list.__module__
'builtins'
```